simphony/io/tests/test_file_mesh.py

```diff
@@ -0,0 +1,603 @@
+
+        self.assertIsNotNone(self.mesh.get_face(fuids[0]))
+
+    def test_add_cell(self):
+        """ Check that a cell can be added correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:4]]
+
+        cell = Cell(puids[0:4])
+
+        cuids = [self.mesh.add_cell(cell)]
+
+        self.assertIsNotNone(self.mesh.get_cell(cuids[0]))
+
+    def test_add_duplicated_point(self):
+        """ Check that a point can be added correctly
+
+        """
+
+        self.mesh.add_point(self.points[0])
+        self.assertRaises(KeyError, self.mesh.add_point, self.points[0])
+
+    def test_add_duplicated_edge(self):
+        """ Check that an edge can be added correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:2]]
+
+        edges = [
+            Edge(puids[0:2])
+        ]
+
+        self.mesh.add_edge(edges[0])
+        self.assertRaises(KeyError, self.mesh.add_edge, edges[0])
+
+    def test_add_duplicated_face(self):
+        """ Check that a face can be added correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:3]]
+
+        faces = [
+            Face(puids[0:3])
+        ]
+
+        self.mesh.add_face(faces[0])
+        self.assertRaises(KeyError, self.mesh.add_face, faces[0])
+
+    def test_add_duplicated_cell(self):
+        """ Check that a cell can be added correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:4]]
+
+        cells = [
+            Cell(puids[0:4])
+        ]
+
+        self.mesh.add_cell(cells[0])
+        self.assertRaises(KeyError, self.mesh.add_cell, cells[0])
+
+    def test_non_emtpy_edges(self):
+        """ Checks that the list of edges is not empty
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:2]]
+
+        edge = Edge(puids[0:2])
+
+        self.mesh.add_edge(edge)
+
+        self.assertTrue(self.mesh.has_edges())
+
+    def test_non_emtpy_faces(self):
+        """ Checks that the list of faces is not empty
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:3]]
+
+        face = Face(puids[0:3])
+
+        self.mesh.add_face(face)
+
+        self.assertTrue(self.mesh.has_faces())
+
+    def test_non_emtpy_cells(self):
+        """ Checks that the list of cells is not empty
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:4]]
+
+        cell = Cell(puids[0:4])
+
+        self.mesh.add_cell(cell)
+
+        self.assertTrue(self.mesh.has_cells())
+
+    def test_get_point(self):
+        """ Check that a point can be retrieved correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:1]]
+
+        point_ret = self.mesh.get_point(puids[0])
+
+        self.assertTrue(isinstance(point_ret, Point))
+        self.assertEqual(puids[0], point_ret.uid)
```

Comment:
You should assert that the points have the expected values, in `coordinates` and `data`